src/main/java/org/broadinstitute/hellbender/utils/codecs/refseq/RefSeqCodec.java

```diff
@@ -0,0 +1,162 @@
+ * You will need to consult individual walkers for the binding name ("refSeqBinding", above)
+ *
+ * <h2>File format example</h2>
+ * If you want to define your own file for use, the format is (tab delimited):
+ * bin, name, chrom, strand, transcription start, transcription end, coding start, coding end, num exons, exon starts, exon ends, id, alt. name, coding start status (complete/incomplete), coding end status (complete,incomplete)
+ * and exon frames, for example:
+ * <pre>
+ * 76 NM_001011874 1 - 3204562 3661579 3206102 3661429 3 3204562,3411782,3660632, 3207049,3411982,3661579, 0 Xkr4 cmpl cmpl 1,2,0,
+ * </pre>
+ * for more information see <a href="http://skip.ucsc.edu/cgi-bin/hgTables?hgsid=5651&hgta_doSchemaDb=mm8&hgta_doSchemaTable=refGene">here</a>
+ * <p>
+ *     
+ * </p>
+ *
+ * @since 2010
+ */
+public class RefSeqCodec extends AsciiFeatureCodec<RefSeqFeature>{
+
+    // codec file extension
+    protected static final String FILE_EXT = "refseq";
+    public static final String HEADER_LINE_CHAR = "#";
+    public static final String LINE_DELIMETER = "\t";
+    /**
+     * The parser to use when resolving genome-wide locations.
+     */
+    private boolean zero_coding_length_user_warned = false;
+
+    public RefSeqCodec() {
+        super(RefSeqFeature.class);
+    }
+
+    @Override
+    public Feature decodeLoc(final LineIterator lineIterator) {
+        final String line = lineIterator.next();
+        if (line.startsWith(HEADER_LINE_CHAR)){
+            return null;
+        }
+        String fields[] = line.split(LINE_DELIMETER);
+        if (fields.length < 3){
+            throw new TribbleException("RefSeq (decodeLoc) : Unable to parse line -> " + line + ", we expected at least 3 columns, we saw " + fields.length);
+        }
+        String contig_name = fields[2];
+        try {
+            return new RefSeqFeature(new SimpleInterval(contig_name, Integer.parseInt(fields[4])+1, Integer.parseInt(fields[5])));
+        //TODO maybe except for malformed simple intervals? Genome locs had that
+        } catch ( NumberFormatException e ) {
+            throw new UserException.MalformedFile("Could not parse location from line: " + line);
+        }
+    }
+
+    /** Fills this object from a text line in RefSeq (UCSC) text dump file */
+    @Override
+    public RefSeqFeature decode(String line) {
+        if (line.startsWith(HEADER_LINE_CHAR)) {
+            return null;
+        }
+        String fields[] = line.split(LINE_DELIMETER);
+
+        // we reference postion 15 in the split array below, make sure we have at least that many columns
+        if (fields.length < 16) {
+            throw new TribbleException("RefSeq (decode) : Unable to parse line -> " + line + ", we expected at least 16 columns, we saw " + fields.length);
+        }
+        String contig_name = fields[2];
+        RefSeqFeature feature = new RefSeqFeature(new SimpleInterval(contig_name, Integer.parseInt(fields[4])+1, Integer.parseInt(fields[5])));
```

Comment:
Ideally you should declare each field index as a named constant at the top of the class (eg., `CONTIG_FIELD_INDEX = 2`, etc.)